test/cli.test.ts

```diff
@@ -0,0 +1,104 @@
+import { exec, ExecException } from 'child_process';
+import * as fs from 'fs';
+import path from 'path';
+import del from 'del';
+
+import { MESSAGES } from '../src/utils';
+
+const GENERATED_FILE_NAME = `test.generated.ts`;
+const tmpDir = 'test/tmp';
+
+function cli(
+  args: string[],
+  cwd: string
+): Promise<{ code: number; error: ExecException | null; stdout: string; stderr: string }> {
+  return new Promise((resolve) => {
+    exec(`ts-node ${path.resolve('./lib/bin/cli.js')} ${args.join(' ')}`, { cwd }, (error, stdout, stderr) => {
```

Comment:
@phryneas I'd like to use the `./src/lib/cli.ts` instead of the built version, but I had troubles getting it execute in the right context and would get module syntax errors 🤦 .

Reply:
```ts
exec(
      `ts-node -T -P ${path.resolve('./tsconfig.json')} ${path.resolve('./src/bin/cli.ts')} ${args.join(' ')}`,
      { cwd },
      (error, stdout, stderr) => {
        resolve({
          error,
          stdout,
          stderr,
        });
```